### PR TITLE
RFD 36: Github Actions

### PR DESCRIPTION
**Pull request by quinqu**

### CODE REVIEW

**Comment by russjones** [comment]
@quinqu Can you either close this PR and open another or update this PR to being it in-sync with what was implemented?

**Review by tcsc — APPROVED** [review]

File: rfd/0036-github-actions.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+| Delete/edit comments on issues or pull requests.   | An attacker could edit or delete important comments. There doesn't seem to be a way to get a malicious commit in master this way.  |
+| Delete logs of a workflow. | An attacker could delete a logs of a workflow that would otherwise fail in their favor. This alone wouldn't be enough to get a malicious commit in for external contributors if the internal contributor that merges their pull request in checks that `Assign` and `Check` workflows pass. Because we do not invalidate reviews for internal contributors when a new commit is pushed and they have the ability to merge their own pull request in, it could be possible to get a malicious commit in. |
+| Re-run a workflow.  | An attacker could re-run a workflow though there wouldn't be any benefit to them even if code was changes. Workflow would just run against the new code and pass/fail accordingly. | 
+| Update an issue or a pull request. | An attacker could update the contents of a pull request or issue.  There doesn't seem to be a way to get a malicious commit in master this way.|
```
Comment (russjones) [inline rfd/0036-github-actions.md:215]: Could an attacker change the contents of any PR?

For example, could an attacker change the contents of an internal PR right before it is merged?

File: rfd/0036-github-actions.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+| Delete/edit comments on issues or pull requests.   | An attacker could edit or delete important comments. There doesn't seem to be a way to get a malicious commit in master this way.  |
+| Delete logs of a workflow. | An attacker could delete a logs of a workflow that would otherwise fail in their favor. This alone wouldn't be enough to get a malicious commit in for external contributors if the internal contributor that merges their pull request in checks that `Assign` and `Check` workflows pass. Because we do not invalidate reviews for internal contributors when a new commit is pushed and they have the ability to merge their own pull request in, it could be possible to get a malicious commit in. |
+| Re-run a workflow.  | An attacker could re-run a workflow though there wouldn't be any benefit to them even if code was changes. Workflow would just run against the new code and pass/fail accordingly. |
```
Comment (russjones) [inline rfd/0036-github-actions.md:214]: Can you explain fail in their favor? Also what are the workflows logs in this context? The checks on the PR would be accurate right?

File: rfd/0036-github-actions.md (hunk 1 above)
Comment (quinqu) [reply rfd/0036-github-actions.md:215]: Correct, but not the actual code, just the description. If you're referring to the actual code, they would need to create a commit and push it to the pull request before the merge. The user would need `contents:write` ( see https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-contents) to post a commit, but we set permissions to none for contents.

File: rfd/0036-github-actions.md (hunk 2 above)
Comment (quinqu) [reply rfd/0036-github-actions.md:214]: Workflow logs are the logs you see with each run in the actions tab. The checks on the PR will be accurate. https://github.blog/changelog/2020-04-21-github-actions-logs-can-now-be-deleted/

**Review by russjones — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]

**Comment by russjones** [comment]
Closing in favor of: https://gravitational.slab.com/posts/using-git-hub-actions-safely-ezuu9pyw.

**Pull request closed without merging**